clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/services/ContainerInformationService.java

```diff
@@ -137,6 +159,22 @@ public boolean taskHasHealthCheck(Service service, String accountName, String re
     return false;
   }
 
+  private String getTargetHealthStatus(EcsTargetHealth ecsTargetHealth) {
+    List<String> statuses =
+        ecsTargetHealth.getTargetHealthDescriptions().stream()
+            .map(tg -> tg.getTargetHealth().getState())
+            .distinct()
+            .collect(Collectors.toList());
+
+    if (statuses.stream().anyMatch(it -> it.equalsIgnoreCase("unhealthy"))) {
+      return "UNHEALTHY";
+    }
+    if (statuses.stream().anyMatch(it -> it.equalsIgnoreCase("healthy"))) {
+      return "HEALTHY";
+    }
+    return "UNKNOWN";
```

Comment:
SO there's a few other states... and the return types here don' seem to match the toPlaformHealthState checsk that do the conversions.  E.g. "HEALTHY" on this needs to match platform state of... "RUNNING" I think to get an "Up" status.  SO not sure this would actually work?

Reply:
This is odd to me as well, I refactor the code to be more cleaner. Basically the targetHealthchecks might have multiple statuses but for this check we only care about "unhealthy" So I refactored my code to only return Down status when unhealthy is found in targetHealthchecks.